tests/src/test/scala/com/nvidia/spark/rapids/ParquetScaleTestSuite.scala

```diff
@@ -0,0 +1,548 @@
+      val cpuFileSize = testDataFile.listFiles(f => f.getName.endsWith(".parquet"))(0).length()
+
+      // write data and get stats on GPU
+      val gpuStats = withGpuSparkSession(getStats(testDataFile), sparkConf)
+      val gpuFileSize = testDataFile.listFiles(f => f.getName.endsWith(".parquet"))(0).length()
+
+      // compare schema
+      if (!skipCheckSchema) {
+        assertResult(cpuStats.schema)(gpuStats.schema)
+      }
+
+      // compare stats
+      assert(cpuStats.rowGroupStats.length == gpuStats.rowGroupStats.length)
+      cpuStats.rowGroupStats.zip(gpuStats.rowGroupStats).foreach {
+        case (cpuRowGroup, gpuRowGroup) => {
+          assert(cpuRowGroup.rowCount == gpuRowGroup.rowCount)
+          assert(cpuRowGroup.columnStats.length == gpuRowGroup.columnStats.length)
+          cpuRowGroup.columnStats.zip(gpuRowGroup.columnStats).foreach {
+            case (cpuColumnStat, gpuColumnStat) => {
+              assert(cpuColumnStat.hasNonNullValue == gpuColumnStat.hasNonNullValue)
+              if (cpuColumnStat.hasNonNullValue) {
+                // compare all the attributes
+                assertResult(cpuColumnStat)(gpuColumnStat)
+              } else {
+                // hasNonNullValue is false, which means stats are invalid, no need to compare
+                // other attributes.
+                /**
+                 * hasNonNullValue means:
+                 *
+                 * Returns whether there have been non-null values added to this statistics
+                 *
+                 * @return true if the values contained at least one non-null value
+                 *
+                 * Refer to link: https://github.com/apache/parquet-mr/blob/apache-parquet-1.10.1
+                 * /parquet-column/src/main/java/org/apache/parquet/column/statistics
+                 * /Statistics.java#L504-L506
+                 *
+                 * e.g.: Spark 31x, for timestamp type
+                 * CPU: hasNonNullValue: false, isNumNullsSet: false, getNumNulls: -1
+                 * GPU: hasNonNullValue: false, isNumNullsSet: true, getNumNulls: 0
+                 *
+                 * Above are expected differences.
+                 */
+                assertResult(cpuColumnStat.primitiveType)(gpuColumnStat.primitiveType)
+              }
+            }
+          }
+        }
+      }
+
+      // Check the Gpu file size is not too large.
+      assert(gpuFileSize < 2 * cpuFileSize)
+
+      (cpuStats, gpuStats)
+    }
+  }
+
+  private val basicTypes = Seq(
+    // "float",  "Blocked by https://github.com/rapidsai/cudf/issues/13948"
+    // "double", "Blocked by https://github.com/rapidsai/cudf/issues/13948"
+    "boolean",
+    "byte",
+    "short",
+    "int",
+    "long",
+    "decimal",
+    "string",
+    "date",
+    "timestamp")
+
+  test("Statistics tests for Parquet files written by GPU, float/double") {
+    assume(false, "Blocked by https://github.com/rapidsai/cudf/issues/13948")
```

Comment:
I'm not sure that this is a bug. According to what is being discussed on the issue CUDF might be doing the right thing when Spark is not doing it. Could we try to update the tests to ignore NaNs? Or at least add in a test that does not have NaN in it?

Reply:
I think it's a bug. Please refer to: https://github.com/rapidsai/cudf/issues/13948#issuecomment-1716869430

Reply:
The can we add a test that does not have nans in it for floating point. Just to verify that we are doing the right thing in those cases too?

Reply:
Please review this follow-up PR: 
https://github.com/NVIDIA/spark-rapids/pull/9256